src/metabase/query_processor/middleware/parameters/mbql.clj

```diff
@@ -88,6 +88,16 @@
      (mbql.u/wrap-field-id-if-needed field)
      (parse-param-value-for-type query param-type param-value (mbql.u/unwrap-field-or-expression-clause field))]))
 
+(defn- update-breakout-unit
+  [query
+   {[_dimension [_field target-field-id {:keys [temporal-unit]}]] :target
+    :keys [value] :as _param}]
+  (let [new-unit (keyword value)]
+    (lib.util.match/replace-in
+      query [:query :breakout]
+      [:field (_ :guard #(= target-field-id %)) (opts :guard #(= temporal-unit (:temporal-unit %)))]
```

Comment:
Is it essential to check for the expected `temporal-unit`?

Reply:
Currently, not really. Soon we plan to allow multiple breakouts for the same column with different units, and this would become a latent bug.

Reply:
Would it be a better world where parameters were attached to specific occurrences of a column (via uuid)?